lib/Sema/CSSimplify.cpp

```diff
@@ -3248,6 +3248,10 @@ ConstraintSystem::matchFunctionTypes(FunctionType *func1, FunctionType *func2,
   if (!matchFunctionIsolations(func1, func2, kind, flags, locator))
     return getTypeMatchFailure(locator);
 
+  if (func1->getLifetimeDependenceInfo() != func2->getLifetimeDependenceInfo()) {
+    return getTypeMatchFailure(locator);
```

Comment:
I think this works because the solver would use a "catch all" conversion fix for arguments but maybe we should add a tailored fix for this to indicate what really mismatched here instead of making developers compare types visually and guess?

Reply:
Are you suggesting to use a `ContextualFailure`? I didn't see precedent in CSSimplify where a `ContextualFailure` was used without attempting a `ConstraintFix`. Most type mismatches without fixits were flagged with the generic `getTypeMatchFailure `. 
If it is okay to use a `ContextualFailure` here, I can do that as a follow on PR.
.

Reply:
I didn't really want to implement a `ConstraintFix ` here. It can cause cascaded SIL diagnostics later on and maybe even more confusing to the user.